Core/ExpectedException/ExpectedExceptionMethodBodyTransformer.cs

```diff
@@ -0,0 +1,97 @@
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+#endregion
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using NUnit2To3SyntaxConverter.Extensions;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+using static NUnit2To3SyntaxConverter.Extensions.SyntaxFactoryUtils;
+
+namespace NUnit2To3SyntaxConverter.ExpectedException
+{
+  public class ExpectedExceptionMethodBodyTransformer : ISyntaxTransformer<MethodDeclarationSyntax, ExpectedExceptionModel>
+  {
+    public MethodDeclarationSyntax Transform (MethodDeclarationSyntax node, ExpectedExceptionModel model)
+    {
+      var indentation = Whitespace (new string (' ', 2));
+      var baseIndentation = node.Body.GetLeadingTrivia();
+      var bodyIndentation = baseIndentation.Add (indentation);
+      var assertThatArgsIndentation = bodyIndentation.Add (indentation).Add (indentation);
+
+      var bodyStatement = node.Body.Statements.Last();
+
+      var lambdaExpression = MakeLambdaExpression (bodyStatement);
+
+      var assertThat = MemberAccess (IdentifierName ("Assert"), "That")
+          .WithLeadingTrivia (bodyIndentation);
+
+
+      var assertThrowsArgumentList = SeparatedList<ArgumentSyntax> (
+          NodeOrTokenList (
+              Argument (lambdaExpression.WithLeadingTrivia (assertThatArgsIndentation)),
+              Token (SyntaxKind.CommaToken).WithTrailingTrivia (Whitespace (Formatting.NewLine)),
+              Argument (model.AsConstraintExpression (assertThatArgsIndentation.ToFullString()))
+              ));
+
+      var assertThatThrows = InvocationExpression (
+          assertThat.WithTrailingTrivia (Whitespace (" ")),
+          ArgumentList (
+              Token (SyntaxKind.OpenParenToken).WithTrailingTrivia (Whitespace (Formatting.NewLine)),
+              assertThrowsArgumentList,
+              Token (SyntaxKind.CloseParenToken)));
+
+      return node.WithBody (
+          node.Body.WithStatements (
+              node.Body.Statements.Replace (
+                  node.Body.Statements.Last(),
+                  ExpressionStatement (assertThatThrows).WithTrailingTrivia (Whitespace (Formatting.NewLine)))));
+    }
+
+    private static LambdaExpressionSyntax MakeLambdaExpression (StatementSyntax bodyStatement)

```

Comment:
We need to pay attention to "one word per concept". Hence we probably shouldn't mix `Create`, `Get`, `Generate`, `Make`